src/uu/ls/src/ls.rs

```diff
@@ -1626,23 +1522,29 @@ fn display_file_name(
 
         if let Some(c) = char_opt {
             name.push(c);
-            width += 1;
         }
     }
 
     if config.format == Format::Long && metadata.file_type().is_symlink() {
         if let Ok(target) = path.read_link() {
-            // We don't bother updating width here because it's not used for long listings
-            let code = if target.exists() { "fi" } else { "mi" };
-            let target_name = color_name(target.to_string_lossy().to_string(), code);
+            // We don't bother updating width here because it's not used for long
+            let mut target_name = target.to_string_lossy().to_string();
+            if let Some(ls_colors) = &config.color {
+                target_name = color_name(&ls_colors, &target, target_name, metadata);
```

Comment:
would it be possible to have a test covering this line?

Reply:
Ah interesting, I just noticed that GNU doesn't actually color the target, so I guess I'll just remove that line:
![image](https://user-images.githubusercontent.com/11643477/115688649-a0cbe300-a35b-11eb-8b55-7e4fc3563e7e.png)